pkg/build/generator/generator.go

```diff
@@ -218,9 +218,11 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
-
+	if request.Binary == nil && bc.Spec.Source.Dockerfile == nil && bc.Spec.Source.Git == nil && len(bc.Spec.Source.Images) == 0 {
```

Comment:
I think my approach allows for a nicer error message  (in validation it's going to need to be a generic "everything is nil, you must specify one" message).  but there's no doubt we were missing that validation, so i'll put it there.